homeassistant/components/flexit/climate.py

```diff
@@ -35,18 +44,25 @@
 SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
 
 
-def setup_platform(hass, config, add_entities, discovery_info=None):
+async def async_setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: DiscoveryInfoType = None,
+):
     """Set up the Flexit Platform."""
     modbus_slave = config.get(CONF_SLAVE)
     name = config.get(CONF_NAME)
     hub = hass.data[MODBUS_DOMAIN][config.get(CONF_HUB)]
```

Comment:
Side note: If integrations should be able to depend on the modbus integration and its hubs we should create a module level function in the modbus integration that returns the correct hub. Integrations are not allowed to directly access `hass.data` to get data from other integrations. The integration, modbus in this case, needs to expose an api for its resources.

Reply:
I will make an extension to modbus exposing the hubs, and use flexit as example, except it later today or tomorrow.

However I think the long term solution is to add hass.get_domain, because there are really no need for integration to hold a list of the hub it "owns".